posthog/performance/sql.py

```diff
@@ -0,0 +1,191 @@
+
+"""
+# expected queries
+
+## get all performance events for a given team's session
+
+allows us to show performance events alongside other logs while viewing a session recording
+
+shard by session id so that when querying for all in session or pageview all data for the results are on the same shard
+
+SELECT * FROM performance_events
+WHERE team_id = 1
+AND session_id = 'my-session-uuid'
+ORDER BY start_time
+
+## get all performance events for a given team's pageview
+
+allows us to show performance events in a waterfall chart for a given pageview
+
+SELECT * FROM performance_events
+WHERE team_id = 1
+AND session_id = 'my-session-uuid'
+AND pageview_id = 'my-page-view-uuid' -- sent by SDK
+ORDER BY start_time
+
+## all other queries are expected to be based on aggregating materialized views built from this fact table
+"""
+
+PERFORMANCE_EVENT_COLUMNS = """
+uuid UUID,
+session_id String,
+window_id String,
+pageview_id String,
+distinct_id String,
+time_origin DateTime64(3, 'UTC'),
+entry_type LowCardinality(String),
+name String,
+team_id Int64,
+current_url String,
+start_time Float64,
+duration Float64,
+redirect_start Float64,
+redirect_end Float64,
+worker_start Float64,
+fetch_start Float64,
+domain_lookup_start Float64,
+domain_lookup_end Float64,
+connect_start Float64,
+secure_connection_start Float64,
+connect_end Float64,
+request_start Float64,
+response_start Float64,
+response_end Float64,
+decoded_body_size Int64,
+encoded_body_size Int64,
+initiator_type LowCardinality(String),
+next_hop_protocol LowCardinality(String),
+render_blocking_status LowCardinality(String),
+response_status Int64,
+transfer_size Int64,
+largest_contentful_paint_element String,
+largest_contentful_paint_render_time Float64,
+largest_contentful_paint_load_time Float64,
+largest_contentful_paint_size Float64,
+largest_contentful_paint_id String,
+largest_contentful_paint_url String,
+dom_complete Float64,
+dom_content_loaded_event Float64,
+dom_interactive Float64,
+load_event_end Float64,
+load_event_start Float64,
+redirect_count Int64,
+navigation_type LowCardinality(String),
+unload_event_end Float64,
+unload_event_start Float64,
+""".strip().rstrip(
+    ","
+)
+
+PERFORMANCE_EVENT_TABLE_ENGINE = lambda: MergeTree("performance_events", replication_scheme=ReplicationScheme.SHARDED)
+
+PERFORMANCE_EVENTS_TABLE_BASE_SQL = """
+CREATE TABLE IF NOT EXISTS {table_name} ON CLUSTER '{cluster}'
+(
+    {columns}
+    {extra_fields}
+) ENGINE = {engine}
+"""
+
+"""
+I think this says store the data on disk grouped by team_id,
+within that group it by session id,
+within that group it by pageview id,
+
+thus in order to query it we are supporting
+
+`where team_id=X`
+
+OR
+
+`where team_id=X and session_id=Y`
+
+OR
+
+`where team_id=X and session_id=Y and pageview_id=Z`
+"""
+PERFORMANCE_EVENTS_TABLE_SQL = (
+    PERFORMANCE_EVENTS_TABLE_BASE_SQL
+    + """PARTITION BY toYYYYMM(time_origin)
+ORDER BY (team_id, session_id, pageview_id)
```

Comment:
so should this be

```
PRIMARY KEY (team_id, session_id, pageview_id)
ORDER BY (team_id, session_id, pageview_id, time_origin)
```

since we want to query by combinations of team, session, and page

but want to also order by time